### PR TITLE
VueJS multiple mount points, `mitt` event bus

### PR DESCRIPTION
**Pull request by gmougeolle**
Addresses etalab/data.gouv.fr#496 : 

- App is now mounted on every DOM node containing the `.vuejs` class
- The same app is mounted on every node
- If the app mount fails, we replace the HTML back to prevent breaking whole blocks
- App now has a `$bus` property allowing to communicate using an event bus (`$bus.emit('event')` and `$bus.on('event')` syntax)
- Freebies : footer DINUM thingy fixed, follow-button adblocker fixed

### CODE REVIEW

**Review by maudetes — COMMENTED** [review]
Nice, it seems quite easy to use this multi apps system!

Could there be some performance impacts to mount the entire app on every node? And if it the case, could we mount specific apps instead (mixing with the logic [here](https://github.com/maudetes/udata-gouvfr/blob/a03233a511bc21b3c703686a58d7ab32a52b1403/theme/js/index.js#L24))?

File: theme/less/components/buttons.less [hunk 1]
```diff
@@ -136,6 +136,7 @@ Because developpers are lazy and less code is better code, there's a `.btn-tab`
     font-weight: bold;
     font-size: @font-size-sm;
     letter-spacing: 0.025em;
+    line-height: 1;
```
[the remark this answers is not in the document]
Comment (gmougeolle) [reply theme/less/components/buttons.less]: Good catch ! This was used to fix the follow-button sizing. However this shouldn't apply to every button. I've moved it to the follow-button file.

**Comment by gmougeolle** [comment]
> Nice, it seems quite easy to use this multi apps system!
> 
> Could there be some performance impacts to mount the entire app on every node? And if it the case, could we mount specific apps instead (mixing with the logic [here](https://github.com/maudetes/udata-gouvfr/blob/a03233a511bc21b3c703686a58d7ab32a52b1403/theme/js/index.js#L24))?

There could be, I couldn't find any literature on the subject. However I can see two arguments for mounting the whole app : 

- Our app is very small so performance shouldn't really be an issue for now (famous last words, future trolls please don't quote me on this)
- With this technique, you're sure that the app contains everything everywhere. It could be confusing mounting a smaller app somewhere in the DOM and then later needing unincluded features only to find them not working. It also allows us to mount app to places where content come from the back-end (while being careful with XSS ;)) and can only be determined at runtime

I think we can go with that solution for now and if performance issues arise, we can easily decide to split the app more in the future. WDYT ?

**Comment by maudetes** [comment]
> I think we can go with that solution for now and if performance issues arise, we can easily decide to split the app more in the future. WDYT ?

Yes agreed, thank you for your inputs!

**Review by maudetes — APPROVED** [review]
:tada: Let's go for multiple VueJS apps !

**Pull request merged**